### PR TITLE
skipping fastq splits for small inputs

### PR DESCRIPTION
**Pull request by khajoue2**
* Small wdl change in Optimus

### CODE REVIEW

File: pipelines/skylab/optimus/Optimus.wdl [hunk 1]
```diff
@@ -86,51 +86,71 @@ workflow Optimus {
...
   call Metrics.CalculateGeneMetrics as GeneMetrics {
     input:
-      bam_input = MergeBam.output_bam
+      bam_input = select_first([STARsoloFastqSingle.bam_output , MergeBam.output_bam])
   }
 
   call Metrics.CalculateCellMetrics as CellMetrics {
     input:
-      bam_input = MergeBam.output_bam,
+      bam_input = select_first([STARsoloFastqSingle.bam_output , MergeBam.output_bam]),
```
Comment (nikellepetrillo) [inline pipelines/skylab/optimus/Optimus.wdl:141]:
```suggestion
      bam_input = select_first([STARsoloFastqSingle.bam_output, MergeBam.output_bam]),
```

Comment (nikellepetrillo) [inline pipelines/skylab/optimus/Optimus.wdl:141]:
```suggestion
      bam_input = select_first([STARsoloFastqSingle.bam_output, MergeBam.output_bam]),
```

File: pipelines/skylab/optimus/Optimus.wdl [hunk 2]
```diff
@@ -86,51 +86,71 @@ workflow Optimus {
...
-  call Merge.MergeSortBamFiles as MergeBam {
-    input:
-      bam_inputs = STARsoloFastq.bam_output,
-      output_bam_filename = output_bam_basename + ".bam",
-      sort_order = "coordinate"
-  }
   call Metrics.CalculateGeneMetrics as GeneMetrics {
     input:
-      bam_input = MergeBam.output_bam
+      bam_input = select_first([STARsoloFastqSingle.bam_output , MergeBam.output_bam])
```
Comment (nikellepetrillo) [inline pipelines/skylab/optimus/Optimus.wdl:136]:
```suggestion
      bam_input = select_first([STARsoloFastqSingle.bam_output, MergeBam.output_bam])
```

File: pipelines/skylab/optimus/Optimus.wdl [hunk 3]
```diff
@@ -86,51 +86,71 @@ workflow Optimus {
...
   }
 
   call Metrics.CalculateCellMetrics as CellMetrics {
     input:
-      bam_input = MergeBam.output_bam,
+      bam_input = select_first([STARsoloFastqSingle.bam_output , MergeBam.output_bam]),
       original_gtf = annotations_gtf
   }
 
+  Array[File] barcodes_out = select_all([STARsoloFastqSingle.barcodes,STARsoloFastq.barcodes])
```
Comment (nikellepetrillo) [inline pipelines/skylab/optimus/Optimus.wdl:145]:
```suggestion
  Array[File] barcodes_out = select_all([STARsoloFastqSingle.barcodes, STARsoloFastq.barcodes])
```

File: pipelines/skylab/optimus/Optimus.wdl [hunk 4]
```diff
@@ -86,51 +86,71 @@ workflow Optimus {
...
+  Array[File] features_out = select_all([STARsoloFastqSingle.features, STARsoloFastq.features])
+  Array[File] matrix_out = select_all([STARsoloFastqSingle.matrix, STARsoloFastq.matrix])
+
   call StarAlign.MergeStarOutput as MergeStarOutputs {
     input:
-      barcodes = STARsoloFastq.barcodes,
-      features = STARsoloFastq.features,
-      matrix = STARsoloFastq.matrix
+      barcodes =  barcodes_out,
```
Comment (nikellepetrillo) [inline pipelines/skylab/optimus/Optimus.wdl:151]:
```suggestion
      barcodes = barcodes_out,
```

**Review by nikellepetrillo — APPROVED** [review]
a few tiny stylistic things, but otherwise looks good to me!

File: pipelines/skylab/optimus/Optimus.changelog.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+# 5.3.1
+2022-04-05 (Date of Last Commit)
+
+* Changed the Optimus wdl logic to skip splitting fastqs if the total input size for read 1 and read 2 is smaller than 30 G.
```
Comment (ekiernan) [inline pipelines/skylab/optimus/Optimus.changelog.md:4]:
```suggestion
* Changed the Optimus WDL logic to skip splitting FASTQs if the total input size for read 1 and read 2 is smaller than 30 GB
```

**Review by ekiernan — APPROVED** [review]
LGTM - one changelog suggestion. Documentation changes related to this PR are on another branch.

**Comment by wjdingman1** [comment]
@khajoue2 can you look at the merge conflicts here?

**Comment by kevinpalis** [comment]
Parking this for now. Closing this PR to avoid too many merge conflicts. The PR is linked to the ticket tracking the work. We will create a new PR when we start working on this again.

**Pull request closed without merging**